pkg/controller/direct/logging/logmetric_controller.go

```diff
@@ -208,6 +208,9 @@ func (a *logMetricAdapter) Update(ctx context.Context, u *unstructured.Unstructu
 		}
 	}
 	if !compareMetricDescriptors(a.desired.Spec.MetricDescriptor, a.actual.MetricDescriptor) {
+		if err := validateImmutableFieldsUpdated(a.desired.Spec.MetricDescriptor, a.actual.MetricDescriptor); err != nil {
```

Comment:
We might not want to enforce this here (in the controller), I'm not sure vs just letting the GCP API call fail....  I'm guessing this is what our controllers do today?

Reply:
Currently the operation fails at DCL level(i.e. DCL might checks for immutable fields too), and won't make API calls. Do we want to check the immutability earlier rather than making calls to GCP?

Reply:
I don't think it's _required_ that we check it here also, but if our controllers do it today then sgtm!